DNN Platform/Library/Entities/Controllers/HostController.cs

```diff
@@ -159,36 +114,25 @@ public int GetInteger(string key, int defaultValue)
             return retValue;
         }
 
-        /// <summary>
-        /// Gets all host settings.
-        /// </summary>
-        /// <returns>host setting.</returns>
-        public Dictionary<string, ConfigurationSetting> GetSettings()
+        /// <inheritdoc/>
+        Dictionary<string, IConfigurationSetting> INewHostController.GetSettings()

```

Comment:
This should be public

Reply:
I extracted the new `IConfigurationSetting` interface into the abstractions project. It is not possible to have 2 identical methods in a class with different return types. The compiler reports errors as it doesn't know how to handle it. Explicit interface implementations like this is a way to solve it. Considering the new interface will only be used for Dependency Injection I believe this change is safe. 

Once we move the deprecated method `Dictionary<string, ConfigurationSetting> GetSettings()` we can remove the explicit declaration and make this public